ghost/webmentions/test/MentionSendingService.test.js

```diff
@@ -0,0 +1,440 @@
+                .post('/webmentions-test')
+                .reply(400);
+
+            const service = new MentionSendingService({externalRequest});
+            await assert.rejects(service.send({
+                source: new URL('https://example.com/source'),
+                target: new URL('https://target.com/target'),
+                endpoint: new URL('https://example.org/webmentions-test')
+            }), /sending failed/);
+            assert(scope.isDone());
+        });
+
+        it('Can handle 500 responses', async function () {
+            const scope = nock('https://example.org')
+                .persist()
+                .post('/webmentions-test')
+                .reply(500);
+
+            const service = new MentionSendingService({externalRequest});
+            await assert.rejects(service.send({
+                source: new URL('https://example.com/source'),
+                target: new URL('https://target.com/target'),
+                endpoint: new URL('https://example.org/webmentions-test')
+            }), /sending failed/);
+            assert(scope.isDone());
+        });
+
+        it('Can handle network errors', async function () {
+            const scope = nock('https://example.org')
+                .persist()
+                .post('/webmentions-test')
+                .replyWithError('network error');
+
+            const service = new MentionSendingService({externalRequest});
+            await assert.rejects(service.send({
+                source: new URL('https://example.com/source'),
+                target: new URL('https://target.com/target'),
+                endpoint: new URL('https://example.org/webmentions-test')
+            }), /network error/);
+            assert(scope.isDone());
+        });
+
+        // Redirects are currently not supported by got for POST requests!
+        //it('Can handle redirect responses', async function () {
+        //    const scope = nock('https://example.org')
+        //        .persist()
+        //        .post('/webmentions-test')
+        //        .reply(302, '', {
+        //            headers: {
+        //                Location: 'https://example.org/webmentions-test-2'
+        //            }
+        //        });
+        //    const scope2 = nock('https://example.org')
+        //        .persist()
+        //        .post('/webmentions-test-2')
+        //        .reply(201);
+        //
+        //    const service = new MentionSendingService({externalRequest});
+        //    await service.send({
+        //        source: new URL('https://example.com'),
+        //        target: new URL('https://example.com'),
+        //        endpoint: new URL('https://example.org/webmentions-test')
+        //    });
+        //    assert(scope.isDone());
+        //    assert(scope2.isDone());
+        //});
+        // TODO: also check if we don't follow private IPs after redirects
+
+        it('Does not send to private IP', async function () {
```

Comment:
Should we comment these out until the `MentionSendingService` is connected?

Reply:
As discussed, these tests can remain here 😊